domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt

```diff
@@ -151,12 +151,16 @@ constructor(
 
     private suspend fun updateMaxFps(currentTargetFrameRate: TargetFrameRate) {
         coroutineScope {
-            val maxFps = cameraProvider.availableCameraInfos.maxOf { cameraInfo ->
-                cameraInfo.supportedFrameRateRanges.maxOf { e ->
-                    e.upper
+            var supportedFixedFrameRates = mutableSetOf<Int>()
+            cameraProvider.availableCameraInfos.forEach() { cameraInfo ->
+                cameraInfo.supportedFrameRateRanges.forEach() { e ->
+                    if (listOf(15,30,60).contains(e.upper)
+                        && e.upper == e.lower) {
```

Comment:
nit: Reverse this to `e.upper == e.lower && listOf(15, 30, 60).contains(e.upper)`.

The comparison operation is much quicker than `contains` and will short-circuit the condition faster when that first comparison is false.

Reply:
nice catch, thanks for pointing this out!